src/tls13.c

```diff
@@ -8519,35 +8530,101 @@ static int SendTls13Certificate(WOLFSSL* ssl)
         /* Cert Req Ctx Len | Cert Req Ctx | Cert List Len | Cert Data Len */
         headerSz = OPAQUE8_LEN + certReqCtxLen + CERT_HEADER_SZ +
                    CERT_HEADER_SZ;
+        XMEMSET(extSz, 0, sizeof(extSz));
+
+        length = 0;
+        listSz = 0;
 
-        ret = TLSX_GetResponseSize(ssl, certificate, &extSz);
+        ret = TLSX_GetResponseSize(ssl, certificate, &extSz[0]);
         if (ret < 0)
             return ret;
 
         /* Create extensions' data if none already present. */
-        if (extSz > OPAQUE16_LEN && ssl->buffers.certExts == NULL) {
-            ret = AllocDer(&ssl->buffers.certExts, extSz, CERT_TYPE, ssl->heap);
+        if (extSz[0] > OPAQUE16_LEN && ssl->buffers.certExts[0] == NULL) {
+            ret = AllocDer(&ssl->buffers.certExts[0], extSz[0],
+                CERT_TYPE, ssl->heap);
             if (ret < 0)
                 return ret;
 
-            extSz = 0;
-            ret = TLSX_WriteResponse(ssl, ssl->buffers.certExts->buffer,
-                                                           certificate, &extSz);
+            extSz[0] = 0;
+            ret = TLSX_WriteResponse(ssl, ssl->buffers.certExts[0]->buffer,
+                certificate, &extSz[0]);
             if (ret < 0)
                 return ret;
         }
+        /* set empty extension as default */
+        for (extIdx = 1; extIdx < (1 + MAX_CERT_EXTENSIONS); extIdx++)
+            extSz[extIdx] = OPAQUE16_LEN;
+
+        /* certificate extension size */
+        totalextSz += extSz[0];
+
+#if defined(HAVE_CERTIFICATE_STATUS_REQUEST)
+        if (ssl->options.side == WOLFSSL_SERVER_END &&
+            ssl->buffers.certChainCnt > 0) {
+            ext = TLSX_Find(ssl->extensions, TLSX_STATUS_REQUEST);
```

Comment:
Put block in own function if it doesn't make a mess of things.

Reply:
Put block into the function